saomsg/msgtoindi.py

```diff
@@ -0,0 +1,528 @@
+                # I am not going to lie. The next line of code 
+                # should be sent back to hell from whence it 
+                # came but it does work. 
+                fxn = lambda val, item=item: hds[item](self, item, val)
+
+                functools.update_wrapper(fxn, hds[item])
+                self.msg_client.subscribe(
+                        item, 
+                        fxn
+                        )
+
+            #build default get and subscribe stuff.
+            value_def = dict(
+                    device=self._devname,
+                    name=item,
+                    label=item,
+                    group="published",
+                    perm='ro', 
+                    state="Idle",
+                    timeout="0",
+                    )
+
+            value_prop = dict(
+                    name=item,
+                    label=item,
+                    value=""
+                    )
+
+            value_vector = self.vectorFactory(
+                    "defTextVector", 
+                    value_def, 
+                    [value_prop]
+                    )
+
+            self.IDDef(value_vector)
+
+            button_def= dict(
+                    device=self._devname,
+                    name=f"{item}_getorsub",
+                    label=item,
+                    group="buttons",
+                    perm='rw', 
+                    state="Idle",
+                    rule="OneOfMany",
+                    timeout="0",
+                    )
+
+            get_switch = dict(
+                    name=f"get_{item}",
+                    label="get",
+                    value=""
+                    ) 
+
+            sub_switch = dict(
+                    name=f"subscribe_{item}",
+                    label="Subscribe",
+                    value=""
+                    )
+
+            button_vector = self.vectorFactory(
+                    "defSwitchVector", 
+                    button_def, 
+                    [get_switch, sub_switch]
+                    )
+            self.IDDef(button_vector)
+        logging.debug("Finished building properties")
+
+
+
+    @device.repeat(1000)
+    async def idle_tasks(self):
+        conn = self.IUFind("CONNECTION")
+        if self.msg_client.running:  
+            if conn["CONNECT"].value == "Off":
+                conn["CONNECT"].value = "On"
+                conn["DISCONNECT"].value = "Off"
+                conn.state="Idle"
+                self.IDSet(conn)
+
+        else:
+            if conn["CONNECT"].value == "On":
+                self.IDMessage("we are not connected to msg_client")
+                conn["CONNECT"].value = "Off"
+                conn["DISCONNECT"].value = "On"
+                conn.state="Idle"
+                self.IDSet(conn)
+          
+
+
+    @classmethod
+    def subscribe(cls, value):
+        def handle_fxn(fxn):
+            return fxn
+
+        return handle_fxn
+
+#    async def repeat_queuer(self):
```

Comment:
is this commented code stuff that's still work in progress?

Reply:
No I believe these were earlier iterations of the classes I was trying to build. I will remove this in the next commit.